### ISSUE TITLE
Unable to research tech requiring Hi-tech tech and beyond

### ISSUE THREAD

**Issue by kawaiiwolf** [jptrrs/HumanResources#26]
Steps to repeat

- New game, loading only [Harmony, Core, Royalty, Hugs, Research Tree, Human Resources] with a Rich Explorer start
- Build a Study Bench & bookshelf [ok]
- Study electronics from library [ok]
- Build a Research Bench [ok]
- Research microelectronics [ok]
- Can't yet build hi-tech research bench
- Go to Study bench an document microelectronics [fail with logged red errors]. Had research jobs queued above it for multi-analyzer/fabrication.
- Moved document microelectronics bill to the top [OK]
- Now able to build hi-tech research bench. Build this + some power generation [ok]
- Queue up multi-analyzer research ... this is where we stall out. Research Tree suggests we need to build a Hi-tech research bench before we can research this. This is where our story ends.

[HumanResourceBugSaves.zip](https://github.com/jptrrs/HumanResources/files/4776555/HumanResourceBugSaves.zip)

**Comment by kawaiiwolf** [comment]
Note: In the attached file, the pawn started with electronics already, so I skipped the first study step and jumped right to researching microelectronics.

Ignore red errors on documenting, I'll figure out explicit steps to reproduce this in another ticket.

**Comment by kawaiiwolf** [comment]
This behavior was seen on both Crash Landed and Rich Explorer scenarios

**Comment by jptrrs** [comment]
If only everyone had this much care reporting a bug! Thanks for the files.
It took me a while to realize the issue can only happen when God Mode is turned off... But I can confirm it now. I'll take a look.

**Comment by MinerSebas** [comment]
I found the problem, but I do not have enough experience with Harmony to know if this can be Patched

Why does it currently not work?
The Game still expects the Building_ResearchBench ThingDef.

Here is an example from ResearchPal, why it displays the Research still as locked:
![Unbenannt](https://user-images.githubusercontent.com/66798382/84606313-696fe880-aea5-11ea-8cba-30b34ebf7a77.PNG)
Here only the two instances Building_ResearchBench need to be replaced with Building_WorkTable through a Transpiler, which is repeated for your ReversePatch of the same Method.

The harder part is the "PlayerHasAnyAppropriateResearchBench" Property of ResearchProjectDef.
As this is a Property and not a method, I do not know how this would have to be done.
![Unbenannt2](https://user-images.githubusercontent.com/66798382/84606900-ebfaa700-aea9-11ea-8dbd-2830326944d7.PNG)
(CanBeResarchedAt was also underlined, as it also expects Building_ResearchBench ).

**Comment by jptrrs** [comment]
That's some solid reverse engineering investigation skills!
Yes, It can be done. And there's multiple ways to go about it.
For the ResearchTree/Pal BuildingPresent, nesting a transpiler in the reverse patch could do it. Another way would be to intercept the result of BuildingPresent with a regular postfix. 
The games's PlayerHasAnyAppropriateResearchBench can be patched directly as a getter method (https://harmony.pardeike.net/articles/annotations.html#getterssetters).
I'm just a little busy these days, and I don't know when I'll have time for this. If you want to give this a shot, I can advise. :-)

**Comment by MinerSebas** [comment]
I am dumb.
I went over the Documentation multiples times, but every time I ignored the Annotations Page. :-(

As mentioned in the PR, fixing the UI will still take some more work, but fixing the fundamental Mechanic had Priority.

**Comment by jptrrs** [comment]
Fixed for the next update.
